Src/Support/GoogleApis.Tests/Apis/Requests/PageStreamerTest.cs

```diff
@@ -0,0 +1,227 @@
+﻿/*
+Copyright 2016 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+using NUnit.Framework;
+
+using Google.Apis.Requests;
+using Google.Apis.Discovery;
+using Google.Apis.Services;
+
+namespace Google.Apis.Tests.Apis.Requests
+{
+    /// <summary>Tests for <see cref="PageStreamer{TResource, TRequest, TResponse, TToken}" />.</summary>
+    [TestFixture]
+    public class PageStreamerTest
+    {
+        private static readonly PageStreamedResource simpleResource = new PageStreamedResource(
+            "simple",
+            new Page(null, 1, 2, 3),
```

Comment:
[optional]
Consider to add a comment here, saying that the first token is null, because it's the first page, and no one refer to it.